File: DP/LeetCode70.go
package dp

func climbStairs(n int) int {
	if n <= 2 {
		return n
	}
	nums := make([]int, n+1)
	nums[0] = 1
	nums[1] = 1
	for i := 2; i <= n; i++ {
		nums[i] = nums[i-1] + nums[i-2]
	}
	return nums[n]
}

File: DP/LeetCode70.cc
#include<iostream>
#include<vector>
using namespace std;

class Solution {
public:

// 时间复杂度在O(n) 空间复杂度也是O(n)
    int climbStairs(int n) {
        if(n <= 2){
            return n;
        }
        vector<int>nums(n+1,1);
        for(int i = 2; i<=n; i++){
            nums[i] = nums[i-1]+nums[i-2];
        }
        return nums[n];
    }
};